### PR TITLE
Convert to Swift3

### PR DESCRIPTION
**Pull request by regnerjr**
Also updated OS X tests so they run the same tests as the iOS Target. 

This depends on a new release of ELFoundation, since the OS X target over there does not build successfully right now.

### CODE REVIEW

File: ELJSBridge/Source/Global.swift [hunk 1]
```diff
@@ -34,67 +34,67 @@ public protocol GlobalSupportable: JSExport {
...
     // Keeping all input values as JSValues so that they can be called on the JS function,  'fn',  in the dispatch and sent back as they came.
     var jsFunctions = [UInt: [JSValue]]()
     
-    public func set(fn: JSValue, timeout: JSValue) -> UInt {
+    open func set(_ fn: JSValue, timeout: JSValue) -> UInt {
         let identifier = UInt(arc4random())
         jsFunctions[identifier] = [fn, timeout]
         
-        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(timeout.toInt32()) * Int64(NSEC_PER_MSEC)), dispatch_get_main_queue()) { () -> Void in
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + Double(Int64(timeout.toInt32()) * Int64(NSEC_PER_MSEC)) / Double(NSEC_PER_SEC)) { () -> Void in
```
Comment (regnerjr) [inline ELJSBridge/Source/Global.swift:62]: This line of code seems a bit 🍌 to me. We got this fancy new `DispatchTime` in Swift 3, and the docs say
>DispatchTime represents a point in time relative to the default clock with nanosecond precision. On Apple platforms, the default clock is based on the Mach absolute time unit.

Is anyone more familiar with this API, What are the units supposed to be here? Maybe this is right if the value of `timeout` is in MSEC?

Let me know what you think. 

Not sure what the migrator was thinking. (The tests all still pass? 🤔 )

File: ELJSBridge/Source/Global.swift [hunk 2]
```diff
@@ -113,25 +113,25 @@ public class Global: NSObject, GlobalSupportable, Scriptable {
 
     }
     
-    internal func setupIntervalDispatch(identifier: UInt, interval: JSValue) {
-        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(interval.toInt32()) * Int64(NSEC_PER_MSEC)), dispatch_get_main_queue()) { () -> Void in
+    internal func setupIntervalDispatch(_ identifier: UInt, interval: JSValue) {
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + Double(Int64(interval.toInt32()) * Int64(NSEC_PER_MSEC)) / Double(NSEC_PER_SEC)) { () -> Void in
```
Comment (regnerjr) [inline ELJSBridge/Source/Global.swift:117]: Same comment here as above.

**Comment by regnerjr** [comment]
🤔 I can't see anything in the travis logs that say why this failed? Any ideas?

**Comment by angelodipaolo** [comment]
Odd that the tests are failing. They seem to be passing when I run them locally and it looks like they only started failing on Travis CI after the change in ed37bb9 which doesn't make any sense.

I'll go ahead and merge. Thanks for taking the time to update this to Swift 3!

**Comment by regnerjr** [comment]
Thanks @angelodipaolo I'm going to keep working on these a little at a time. Native Grocery for iOS is using them and we're not ready to got to Swift 3 yet. But I'd like to be ready with these when we do. Thanks again for the help, and support.

**Pull request merged**